spec/requests/providers_spec.rb

```diff
@@ -421,6 +421,25 @@ def have_endpoint_attributes(expected_hash)
         .and_return([OvirtSDK4::ProbeResult.new(:version => '3')])
     end
 
+    it 'allows provider specific attributes to be specified' do
+      allow(ManageIQ::Providers::Azure::CloudManager).to receive(:api_allowed_attributes).and_return(%w[azure_tenant_id])
+      tenant = FactoryGirl.create(:cloud_tenant)
+      api_basic_authorize collection_action_identifier(:providers, :create)
+
+      post(api_providers_url, :params => { "type"            => "ManageIQ::Providers::Azure::CloudManager",
+                                           "name"            => "sample azure provider",
+                                           "hostname"        => "hostname",
+                                           "zone"            => @zone,
+                                           "azure_tenant_id" => tenant.id,
+                                           "credentials"     => {}})
+
+      expected = {
+        "results" => [a_hash_including("uid_ems" => tenant.id.to_s, "name" => "sample azure provider")]
+      }
```

Comment:
@jntullo - does the `results` hash mirror the database columns? Is that why the first hash key is `uid_ems` and not `azure_tenant_id`

Reply:
@bronaghs yeah, it is all of the physical attributes